### DISCUSSION TITLE
How to find root cause span and alerting in opentelemerty trace with Clickhouse?

### DISCUSSION THREAD

**Discussion by hueiyuan** [grafana/clickhouse-datasource#718] [Q&A]
## Description
Hi, community.
We have the question about opentelemerty trace span. In our scenario, one span can be represent to one endpoint call. For example:
one of trace is: A -> B -> C
another trace is: A -> D

If A this span have problem so that impact on downstream span problem. How do we find the root cause upstream span and execute alert to corresponding user.
In this scenario, will trigger 2 alerts. But if trace becomes larger, will trigger n alerts. We hope just notify one alert message to shows the root span. (just show A span related information)
Does anyone have ideas to solve about it? Thank you.

**Comment by SpencerTorres** [comment]
I think it would help to see a query + screenshot + anonymized example data for this particular issue. It seems like this could be solved with a filter though. If a single filter doesn't work, then the query might need to be written with the SQL Editor.

If the query is hard to rewrite every time you open the explore page, you can bookmark the URL to have it as a template. You can also star the query from the query history panel.
